kornia/contrib/connected_components.py

```diff
@@ -34,8 +35,8 @@ def connected_components(image: Tensor, num_iterations: int = 100) -> Tensor:
     if not isinstance(image, Tensor):
         raise TypeError(f"Input imagetype is not a Tensor. Got: {type(image)}")
 
-    if not isinstance(num_iterations, int) or num_iterations < 1:
-        raise TypeError("Input num_iterations must be a positive integer.")
+    if not isinstance(num_iterations, int) or num_iterations < 0:
```

Comment:
```suggestion
    if not isinstance(num_iterations, int) or num_iterations <= 0:
```
non-zero and positive numbers

Reply:
I use `num_iterations=0` to detect if I need to stop when reaching convergence or maximum iterations.

Reply:
At the moment I left the default value to 100 for backward compatibility but maybe it could be better to set it to 0 since it ensures that the algorithm works correctly.